Shell/main.cpp

```diff
@@ -147,6 +147,43 @@ static int sh_unset(int argc, char** argv)
     return 0;
 }
 
+static String expanded_tilde_expression_from(char* arg)
+{
+    int len = strlen(arg);
+    ASSERT(len);
+
+    StringBuilder login_name;
+    int first_slash_index = len;
+    for (int i = 1; i < len; ++i) {
+        if (arg[i] == '/') {
+            first_slash_index = i;
+            break;
+        }
+        login_name.append(arg[i]);
+    }
+
+    StringBuilder path;
+    for (int i = first_slash_index; i < len; ++i)
+        path.append(arg[i]);
+
+    if (login_name.length() == 0) {
+        char* home = getenv("HOME");
```

Comment:
`const char*`

Reply:
Every time you use `char*` (as opposed to `const char*`), it's screaming to me — "attention! this is not just a string! I'm going to mutate this!". So if you don't mean that, use `const char*` (or languages where `const` is the default :wink:)

Reply:
Thanks 🙂. Yeah, I need to get better at not forgetting to add const